salt/_grains/etcd_info.py

```diff
@@ -0,0 +1,40 @@
+try:
+    import etcd
+    HAS_LIBS = True
+except ImportError:
+    HAS_LIBS = False
+
+from salt.exceptions import CommandExecutionError
+
+def etcd_info():
+
+  etcd_info = {}
+  etcd_info['members_all'] = []
+  etcd_info['member_type'] = "proxy"
+
+  if HAS_LIBS:
+    etcd_info['available'] = True
+    client = etcd.Client(host='127.0.0.1', port=2379)
+  else:
+    etcd_info['available'] = False
+    raise CommandExecutionError(
+        '(unable to import etcd, '
+        'module most likely not installed)'
+    )
+
+  with open("/etc/machine-id", "r") as machine_file:
+    machine_id = machine_file.read().rstrip()
+  machine_file.close()
+
+  etcd_members = client.members
```

Comment:
This will be handled by python-etcd exception, we will get following output
```
[WARNING ] Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f558c7a2b50>: Failed to establish a new connection: [Errno 111] Connection refused',)': /v2/members
[WARNING ] Retrying (Retry(total=1, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f558c7a2cd0>: Failed to establish a new connection: [Errno 111] Connection refused',)': /v2/members
[WARNING ] Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f558c7a2e10>: Failed to establish a new connection: [Errno 111] Connection refused',)': /v2/members
[ERROR   ] Request to server http://127.0.0.1:2379 failed: MaxRetryError("HTTPConnectionPool(host='127.0.0.1', port=2379): Max retries exceeded with url: /v2/members (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f558c7a2f10>: Failed to establish a new connection: [Errno 111] Connection refused',))",)
[CRITICAL] Failed to load grains defined in grain file etcd_info.etcd_info in function <function etcd_info at 0x7f558c7c8140>, error:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 723, in grains
    ret = fun()
  File "/var/cache/salt/minion/extmods/grains/etcd_info.py", line 32, in etcd_info
    etcd_members = client.members
  File "/usr/lib/python2.7/site-packages/etcd/client.py", line 341, in members
    raise etcd.EtcdException("Could not get the members list, maybe the cluster has gone away?")
EtcdException: Could not get the members list, maybe the cluster has gone away
```